dbfilter_from_header/override.py

```diff
@@ -6,24 +6,99 @@
 import logging
 import re
 
+import psycopg2
+
+import odoo.service.db
 from odoo import http
+from odoo.http import get_default_session, root
 from odoo.tools import config
 
+_logger = logging.getLogger(__name__)
+
 db_filter_org = http.db_filter
 
 
-def db_filter(dbs, httprequest=None):
-    dbs = db_filter_org(dbs, httprequest)
+def db_filter(dbs, host=None, httprequest=None):
+    dbs = db_filter_org(dbs, host)
     httprequest = httprequest or http.request.httprequest
     db_filter_hdr = httprequest.environ.get("HTTP_X_ODOO_DBFILTER")
     if db_filter_hdr:
         dbs = [db for db in dbs if re.match(db_filter_hdr, db)]
     return dbs
 
 
+# we don't rely on original http.db_list
+# because it calls db_filter(dbs, host)
+# while we also need to pass httprequest
+def db_list(force=False, host=None, httprequest=None):
+    try:
+        dbs = odoo.service.db.list_dbs(force)
+    except psycopg2.OperationalError:
+        return []
+    return db_filter(dbs, host, httprequest)
```

Comment:
I propose to use keyword arguments here, so

```suggestion
    return db_filter(dbs, host=host, httprequest=httprequest)
```